components/brave_rewards/resources/extension/brave_rewards/twitter.ts

```diff
@@ -0,0 +1,158 @@
+  braveTipIcon.style.display = 'inline-block'
+  braveTipIcon.style.fontSize = '18px'
+  braveTipIcon.style.fontStyle = 'normal'
+  braveTipIcon.style.height = '16px'
+  braveTipIcon.style.marginTop = '5px'
+  braveTipIcon.style.position = 'relative'
+  braveTipIcon.style.verticalAlign = 'baseline'
+  braveTipIcon.style.width = '16px'
+  braveTipIconContainer.appendChild(braveTipIcon)
+
+  // Create the tip action count (typically used to present a counter
+  // associated with the action, but we'll use it to display a static
+  // action label)
+  const braveTipActionCount = document.createElement('span')
+  braveTipActionCount.className = 'ProfileTweet-actionCount'
+  braveTipActionCount.style.color = '#657786'
+  braveTipActionCount.style.display = 'inline-block'
+  braveTipActionCount.style.fontSize = '12px'
+  braveTipActionCount.style.fontWeight = 'bold'
+  braveTipActionCount.style.lineHeight = '1'
+  braveTipActionCount.style.marginLeft = '6px'
+  braveTipActionCount.style.position = 'relative'
+  braveTipActionCount.style.verticalAlign = 'text-bottom'
+  braveTipButton.appendChild(braveTipActionCount)
+
+  // Create the tip action count presentation
+  const braveTipActionCountPresentation = document.createElement('span')
+  braveTipActionCountPresentation.className = 'ProfileTweet-actionCountForPresentation'
+  braveTipActionCountPresentation.textContent = getMessage('twitterTipsIconLabel')
+  braveTipActionCount.appendChild(braveTipActionCountPresentation)
+
+  // Create the shadow DOM root that hosts our injected DOM elements
+  const shadowRoot = braveTipAction.attachShadow({ mode: 'open' })
+  shadowRoot.appendChild(braveTipButton)
+
+  return braveTipAction
+}
+
+const configureBraveTipAction = () => {
+  chrome.runtime.sendMessage('rewardsEnabled', function (rewards) {
+    const msg = {
+      type: 'inlineTipSetting',
+      key: 'twitter'
+    }
+    chrome.runtime.sendMessage(msg, function (inlineTip) {
+      const tweets = document.getElementsByClassName('tweet')
+      for (let i = 0; i < tweets.length; ++i) {
+        const actions = tweets[i].getElementsByClassName('js-actions')[0]
+        if (actions) {
+          const braveTipActions = actions.getElementsByClassName('action-brave-tip')
+          const tippingEnabled = rewards.enabled && inlineTip.enabled
+          if (tippingEnabled && braveTipActions.length === 0) {
+            actions.appendChild(createBraveTipAction(tweets[i]))
+          } else if (!tippingEnabled && braveTipActions.length === 1) {
+            actions.removeChild(braveTipActions[0])
+          }
+        }
+      }
+    })
+  })
+  timeout = setTimeout(configureBraveTipAction, 3000)
```

Comment:
I think you're missing a line to declare the `timeout` variable

Reply:
I had to move it to the top of the file, I think lint was complaining that it was being used before it was defined:

https://github.com/brave/brave-core/pull/2195/files#diff-31fb8eabba7a25b47b33d069f907d3faR8

Reply:
actually, found it!